src/aggregator/aggregator/generic_elem.go

```diff
@@ -123,84 +130,59 @@ type GenericElem struct {
 	elemBase
 	typeSpecificElemBase
 
-	values              []timedAggregation         // metric aggregations sorted by time in ascending order
-	toConsume           []timedAggregation         // small buffer to avoid memory allocations during consumption
-	lastConsumedAtNanos int64                      // last consumed at in Unix nanoseconds
-	lastConsumedValues  []transformation.Datapoint // last consumed values
+	values    []timedAggregation // metric aggregations sorted by time in ascending order
+	toConsume []timedAggregation // small buffer to avoid memory allocations during consumption
+	// map of the previous consumed values for each timestamp in the buffer. needed to support binary transforms that
+	// need the value from the previous timestamp.
+	consumedValues map[int64][]transformation.Datapoint
```

Comment:
nit: Can use `xtime.UnixNano` here so there are helper functions to go from time / to time, etc and improve readability a slight amount *shrug*.